src/EachPromise.php

```diff
@@ -80,9 +80,9 @@ public function promise(): PromiseInterface
         }
 
         try {
+            $this->iterable->rewind();
```

Comment:
Flipped the ordering here so that createPromise() can check whether the iterable is empty.